soccer/gameplay/GameplayModule.cpp

```diff
@@ -314,6 +314,9 @@ void Gameplay::GameplayModule::run() {
     _ballMatrix = Geometry2d::TransformMatrix::translate(
         _context->world_state.ball.position);
 
+    _context->globalObstacles = globalObstacles();
+    _context->goalZoneObstacles = goalZoneObstacles();
```

Comment:
Does it make sense to have the global obstacles set here? It doesn't really feel like it fits. For the sake of this PR though, this is fine

Reply:
Yeah this has always been where they're created but I think it really belongs with the field geometry calculations.